imx/signers/stark/factory.go

```diff
@@ -24,9 +27,15 @@ func GenerateKey() (*big.Int, error) {
 			return nil, err
 		}
 	}
-	return curve.GetRandomPrivateKey()
+	privKey, err := ecdsa.GenerateKey(curve, rand.Reader)
```

Comment:
Check that the `crypto/rand` guarantees secure random generation

https://cryptobook.nakov.com/secure-random-generators/secure-random-generators-csprng